src/sentry_envelope.c

```diff
@@ -614,6 +338,21 @@ sentry__envelope_serialize_into_stringbuilder(
     }
 }
 
+char *
+sentry_envelope_serialize_consume(sentry_envelope_t *envelope, size_t *size_out)
```

Comment:
Is there a reason to have this function consume? It seems a bit restrictive. Worst case, the caller can still call free.

Reply:
since the previous code was using callbacks, it could reuse the buffer of a raw envelope. otherwise we would have to make a copy.